NYPL-discovery/discovery-front-end
# SCC-3466: Bib page item pagination and view all update

**What's this do?**
Resolves [SCC-3466](https://jira.nypl.org/browse/SCC-3466).

This removes all remnants of client-side filtering or pagination on a bib page. The only exception is when then "view all items" button is clicked and client-side requests are made to incrementally fetch more items for a bib. 

This is done by setting the hash `#view-all-items` in the URL. When a page with this hash is shared, the client-side incremental item fetching is performed. This also means that the state for the the `ItemsContainer` component is slightly cleaner because now it only needs to care about the `js` value. All other values have been removed or moved to the parent.

When requesting the next or previous set of items, all previous items are removed and replaced with the next batch of items. This means that `ItemsContainer` does not need to keep track of items in its state anymore nor deal with React lifecycle functions in the component.

One big update is `itemPage` is now `item_page` to be consistent with the rest of the URL query params.

**Why are we doing this? (w/ JIRA link if applicable)**
[Quick blurb about why the code is needed and Jira link/ Do these changes meet the business requirements of the story?]

**Do these changes have automated tests?**
Yes

**How should this be QAed?**
Review multiple bib pages.
1. Test out the "View all items" button.
2. Test the pagination. Each new page (next or previous) should make an API call and display the next or previous 20 items).
3. Test the filters.
4. Test the filters _and_ the pagination.
5. Test the filters _and_ clicking on the "view all items" button.

**Dependencies for merging? Releasing to production?**
[Description of any watchouts, dependencies, or issues we should be aware of]

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**
